### PR TITLE
[Homepage] Implement UI for Homepage

### PR DESCRIPTION
**Pull request by BlueDaisy061**
### Description:
- Add SVG and png images on the Homepage
- Implement UI for the Home page
- Create Typography components for all pages.
- Create spacer component.

### Screenshot:
No screenshots due to the very long information. Please pull it down and run it in local.

### CODE REVIEW

File: src/components/Typography/index.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Heading1 from "./Heading1";
+import Heading2 from "./Heading2";
+import Heading3 from "./Heading3";
+import Caption from "./Caption";
+import Paragraph from "./Paragraph";
+import SubHeading from "./SubHeading";
+
+export {Heading1, Heading2, Heading3, Caption, Paragraph, SubHeading};
```
Comment (tainguy22112000) [inline src/components/Typography/index.tsx:8]: or
`export * from "./Heading1"`,
but it's OK in this way.

File: src/App.tsx [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import { RouterProvider } from 'react-router-dom'
 import { router } from './routes'
-import { Footer } from './components/Footer'
-import Header from './components/Header'
 
 function App() {
   return (
-    <div>
+    <div style={{backgroundColor: "#FBFBFB"}}>
```
Comment (tainguy22112000) [inline src/App.tsx]: consider that this color could be reused.

File: src/components/Card/index.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import styles from './styles.module.scss'
+
+type FeatureProps = {
+    children: ReactNode;
+    padding: string;
+}
+
+export default function Card({ children, padding }: FeatureProps) {
+  return (
+    <div className={classNames(styles.container, {
```
Comment (tainguy22112000) [inline src/components/Card/index.tsx:12]: good job babii ~~

File: src/components/Card/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { ReactNode } from 'react'
+import classNames from 'classnames';
+import styles from './styles.module.scss'
+
+type FeatureProps = {
```
Comment (tainguy22112000) [inline src/components/Card/index.tsx]: Use consistent naming for component props types
CardProps

File: src/components/Card/index.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    children: ReactNode;
+    padding: string;
+}
+
+export default function Card({ children, padding }: FeatureProps) {
```
Comment (tainguy22112000) [inline src/components/Card/index.tsx]: highly recommend passing default param
padding =. "md"

File: src/components/Spacer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import classNames from 'classnames';
+import styles from './styles.module.scss'
+
+type SpacerProps = {
+    size: string;
+}
+
+export default function Spacer({size}: SpacerProps) {
```
Comment (tainguy22112000) [inline src/components/Spacer/index.tsx]: same above

File: src/components/UserInfo/index.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+type UserAvatarProps = {
+    avatar: string,
+    name: string,
+    role: string
+}
+
+export default function UserInfo({avatar, name, role}: UserAvatarProps) {
+    return (
+    <div className={styles.container}>
+        <img src={avatar} />
```
Comment (tainguy22112000) [inline src/components/UserInfo/index.tsx]: add alt for image

File: src/components/UserInfo/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,20 @@
+@use 'src/styles' as *;
+.container {
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-start;
+    align-items: center;
+    .userInfo {
+        margin-left: 12px;
+        .userName {
```
Comment (tainguy22112000) [inline src/components/UserInfo/styles.module.scss:9]: prevent nesting more than 3 levels

File: src/page/home/index.tsx [hunk 9]
```diff
@@ -1,13 +1,157 @@
...
+              <div className={styles.solutionDetails} key={key}>
+                <span className={styles.solutionIcon}>{value.icon}</span>
+                <span>{value.name}</span>
+              </div>
+            ))}
+          </div>
+          <div>
+            <div className={styles.imgStructure}>
+              <div className={styles.imgLeft}>
+                <img className={styles.imgFloatRight} src="src\assets\images\Solution1.png"/>
```
Comment (tainguy22112000) [inline src/page/home/index.tsx]: alt

File: src/page/home/index.tsx [hunk 10]
```diff
@@ -1,13 +1,157 @@
...
+                <Spacer size="2xl"/>
+                <img src="src\assets\images\Solution2.png"/>
+              </div>
+              <div className={styles.imgRight}>
+                <img src="src\assets\images\Solution3.png"/>
+                <Spacer size="2xl"/>
+                <img src="src\assets\images\Solution4.png"/>
```
Comment (tainguy22112000) [inline src/page/home/index.tsx]: alt

File: src/page/home/index.tsx [hunk 11]
```diff
@@ -1,13 +1,157 @@
...
+              </div>
+            </div>
+            <Spacer size="2xl"/>
+            <div className={styles.imgStructure}>
+              <img className={styles.imgMarginLeft} src="src\assets\images\Solution5.png" />
```
Comment (tainguy22112000) [inline src/page/home/index.tsx]: same above

File: src/components/Typography/index.tsx (hunk 1 above)
Comment (tainguy22112000) [reply src/components/Typography/index.tsx:8]: another way
const Heading1: FC<TypoProps> = ({children} ) => {
     return (
           <div className={styles.subheading}>{children}</div>)
}
const Heading2: FC<TypoProps> = ({children} ) => {
     return (
           <div className={styles.subheading}>{children}</div>)
}
const Heading3: FC<TypoProps> = ({children} ) => {
     return (
           <div className={styles.subheading}>{children}</div>)
}

...

export {Heading1, Heading2, Heading3, ...}

File: src/page/home/index.tsx [hunk 12]
```diff
@@ -1,13 +1,157 @@
...
+          </div>
+          <div>
+            <div className={styles.imgStructure}>
+              <div className={styles.imgLeft}>
+                <img className={styles.imgFloatRight} src="src\assets\images\Solution1.png"/>
+                <Spacer size="2xl"/>
+                <img src="src\assets\images\Solution2.png"/>
+              </div>
+              <div className={styles.imgRight}>
+                <img src="src\assets\images\Solution3.png"/>
```
Comment (tainguy22112000) [inline src/page/home/index.tsx]: alt

File: src/page/home/index.tsx [hunk 13]
```diff
@@ -1,13 +1,157 @@
...
+                <Spacer size="lg" />
+                <Paragraph>{value.content}</Paragraph>
+              </Card>
+            ))}
+        </div>
+      </div>
+
+      {/* Logos */}
+      <div className={styles.logos}>
+          <div>
```
Comment (tainguy22112000) [inline src/page/home/index.tsx:123]: alt

File: src/page/home/index.tsx [hunk 14]
```diff
@@ -1,13 +1,157 @@
...
+                <input type="text" placeholder='Enter your name' className={styles.formInput} />
+                <Spacer size="xl"/>
+                <input type="text" placeholder='Enter your name' className={styles.formInput} />
+                <Spacer size="xl"/>
+                <textarea name="form-message" id="form-id" rows={6} placeholder='Your message' className={styles.formInput}></textarea>
+              </form>
+              <PrimaryButton title="Confirm"/>
+          </div>
+          <div>
+            <img src="src\assets\images\MoreInfoImg.png" />
```
Comment (tainguy22112000) [inline src/page/home/index.tsx]: alt

File: src/page/home/styles.module.scss [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+.title {
+    margin: 70px 0px;
+}
+
+.video {
+    width: 70%;
+    display: block;
+    margin-left: auto;
```
Comment (tainguy22112000) [inline src/page/home/styles.module.scss]: margin: 0 auto 14px

File: src/page/home/styles.module.scss [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+            flex-direction: column;
+            margin: 40px 0;
+            .formInput {
+                padding: 16px;
+                color: $color-grey;
+                border: 1px solid $color-lightred;
+                border-radius: 10px;
+                 :focus {
+                    border: 1px solid $color-lightred;
+                    box-shadow: rgba(255, 255, 255, 255);
```
Comment (tainguy22112000) [inline src/page/home/styles.module.scss]: rgba(red, green, blue, alpha)
alpha 0 - 1
this is $color-white for box-shadow ? pls check it again

File: src/App.tsx (hunk 2 above)
Comment (BlueDaisy061) [reply src/App.tsx]: **Call for help:** how to import sass varibles to use as an inline style? I mean not to create another style.module file but use it directly in this component.

File: src/components/Card/index.tsx (hunk 4 above)
Comment (BlueDaisy061) [reply src/components/Card/index.tsx]: Updated.

File: src/components/Card/index.tsx (hunk 5 above)
Comment (BlueDaisy061) [reply src/components/Card/index.tsx]: Sure. Thanks for your notice.

File: src/page/home/styles.module.scss (hunk 16 above)
Comment (BlueDaisy061) [reply src/page/home/styles.module.scss]: `textarea:focus,
input:focus {
outline: none;
box-shadow: 0px 8px 32px rgba(177, 177, 177, 0.5);}`
I mean this one. Forgot to remove it after testing it unsuccessfully.

File: src/components/UserInfo/styles.module.scss (hunk 8 above)
Comment (BlueDaisy061) [reply src/components/UserInfo/styles.module.scss:9]: it actually just equals 3 levels, no more. But I have fixed 4-level nested in `src\page\Home\styles.module.scss`, you can check on [f344cf5](https://github.com/tainguy22112000/DaisyAgency/pull/3/commits/f344cf5cebffe06265beeb9440cab23ee15f9d9d)

File: src/App.tsx (hunk 2 above)
Comment (tainguy22112000) [reply src/App.tsx]: you cannot directly import SCSS variables into your components. The possible way is to create a new style.module file, create scss variables, and import it.

**Comment by tainguy22112000** [comment]
all img tags must have "alt" attribute. pls check it again ~

**Comment by BlueDaisy061** [comment]
> all img tags must have "alt" attribute. pls check it again ~

I'm sure I did.

File: src/App.tsx (hunk 2 above)
Comment (BlueDaisy061) [reply src/App.tsx]: Please check the new commit [ac9985a](https://github.com/tainguy22112000/DaisyAgency/pull/3/commits/ac9985ade663388d5114a8527d2a109e87fe62a8)

**Pull request merged**